### PR TITLE
Fixed bug when trying to open a row with a binary encoded uuid as PK

### PR DESCRIPTION
**Pull request by VictorMinemu**
## Change Summary

Provide summary of changes with issue number if any.

## Change type

- [ ] feat: (new feature for the user, not a new feature for build script)
- [x] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.
Added a check on the WherePK constructor to check if the data type on the db is binary(16) and if the pk is a valid uuid.
If it's a uuid then returns it as a Binary.

## Additional information / screenshots (optional)

Anything for maintainers to be made aware of

### CODE REVIEW

**Comment by VictorMinemu** [comment]
Added the fix for the equal date filter with columns of DB type timestamp.

**Review by pranavxc — APPROVED** [review]

**Comment by pranavxc** [comment]
re #6023

File: packages/nocodb/src/db/conditionV2.ts [hunk 1]
```diff
@@ -419,6 +419,8 @@ const parseConditionV2 = async (
                 ].includes(column.uidt)
               ) {
                 qb = qb.where(field, val);
+              } else if (column.ct == 'timestamp') {
```
Comment (wingkwong) [inline packages/nocodb/src/db/conditionV2.ts:422]: `column.ct === 'timestamp'`

File: packages/nocodb/src/db/BaseModelSqlv2.ts [hunk 2]
```diff
@@ -3966,7 +3966,21 @@ function _wherePk(primaryKeys: Column[], id) {
...
+      primaryKeys[i].ct === 'binary(16)' &&
+      ((ids[i] + '').length === 36 || (ids[i] + '').length === 32)
+        ? (ids[i] + '').length === 32
+          ? (ids[i] + '').replace(
+              /(.{8})(.{4})(.{4})(.{4})(.{12})/,
+              '$1-$2-$3-$4-$5',
+            )
+          : ids[i] + ''
+        : null;
+
```
[the remark this answers is not in the document]
Comment (wingkwong) [reply packages/nocodb/src/db/BaseModelSqlv2.ts:3980]: prefer Pranav's version as the readability is much better.

**Review by wingkwong — CHANGES_REQUESTED** [review]

File: packages/nocodb/src/db/BaseModelSqlv2.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (VictorMinemu) [reply packages/nocodb/src/db/BaseModelSqlv2.ts]: I agree it's much more readable.
Fixed in the last commit

**Pull request merged**